packages/tabs/src/Tabs.tsx

```diff
@@ -163,105 +173,116 @@ function Tabs({
     ? setControlledSelected
     : setUncontrolledSelected;
 
-  function handleChange(
-    e: React.SyntheticEvent<Element, MouseEvent>,
-    index: number,
-  ) {
-    setSelected(index);
-  }
+  const handleChange = useCallback(
+    (e: React.SyntheticEvent<Element, MouseEvent>, index: number) => {
+      setSelected(index);
+    },
+    [setSelected],
+  );
 
-  const getEnabledIndexes: () => [Array<number>, number] = () => {
+  const getEnabledIndexes: () => [Array<number>, number] = useCallback(() => {
     const enabledIndexes = childrenArray
       .filter(child => !child.props.disabled)
       .map(child => childrenArray.indexOf(child));
 
     return [enabledIndexes, enabledIndexes.indexOf(selected!)];
-  };
-
-  const handleArrowKeyPress = (e: KeyboardEvent) => {
-    if (!(e.metaKey || e.ctrlKey)) {
-      if (e.keyCode === keyMap.ArrowRight) {
-        const [enabledIndexes, current] = getEnabledIndexes();
-        setSelected(enabledIndexes[(current + 1) % enabledIndexes.length]);
-      } else if (e.keyCode === keyMap.ArrowLeft) {
-        const [enabledIndexes, current] = getEnabledIndexes();
-        setSelected(
-          enabledIndexes[
-            (current - 1 + enabledIndexes.length) % enabledIndexes.length
-          ],
-        );
+  }, [childrenArray, selected]);
```

Comment:
this will still be re-evaluated on every render because `childrenArray` is not memoized

Reply:
👍